pkg/gather/tasks_processing.go

```diff
@@ -0,0 +1,73 @@
+package gather
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/openshift/insights-operator/pkg/gather/common"
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// TODO: docs
+
+type Task struct {
+	Name    string
+	F       common.GatheringClosure
```

Comment:
I don't know what the Go convention is, but calling it `F` looks to me quite cryptic. What about call it `Run` ?

Reply:
My thoughts were that it's a function pointer, so `F`, but can be `Run` to, why not